### PR TITLE
web: display toast notification on uibutton click

### PR DESCRIPTION
**Pull request by landism**
### Problem

When clicking a UIButton, the only feedback that anything happened is some logging of the command and the output. That logging goes to the resource log, or if it's a header button, only to the global log.
If the user does not have the appropriate log pane open, they might wonder if Tilt even recognized the button click or the command ran at all.

### Solution

Display a toast notification when the user clicks a UIButton, with a link to the appropriate log view
image

NB: I originally tried implementing this using MUI's [snackbar](https://material-ui.com/components/snackbars/) directly, but it was kind of annoying in that all the buttons would try to render their snackbars on top of each other (it wasn't always the most recent on top, and if a shorter one was on top of a longer one, you could still see the uncovered part of the longer one underneath) so we'd have to add a context to allow one snackbar to replace another. MUI's snackbar documentation [suggests](https://material-ui.com/components/snackbars/#notistack) the [notistack](https://github.com/iamhosseindhv/notistack) library, which basically just gives you this context provider (though MUI takes care to point out that the material design spec discourages having multiple snackbars - maybe it'd be better to limit to 1 at a time?).

### CODE REVIEW

**Review by lizzthabet — COMMENTED** [review]
tested locally without any issues! i think my biggest complaint is that the notifications feel out of place in the app, both from a style and ux perspective. i think this could be mitigated by either:

* giving some lightweight style to the notifications, so they at least use our colors and fonts (maybe they could look a bit like tooltips?)

* revisting the ux of the button feedback to resemble the `copy to clipboard` behavior (where the button content could temporarily be replaced with a ✅ `Triggered` after success. the button trigger lifecycle could look like: click -> loading state -> brief success message -> back to normal.)

i know you already chatted with surbhi about this being okay to ship as-is, so feel free to ignore my bigger picture feedback or ask others to chime in.

**Comment by landism** [comment]
> revisting the ux of the button feedback to resemble the copy to clipboard behavior

I like this a lot! But I'm leaning against dragging this out longer for now.

File: web/src/ApiButton.tsx [hunk 1]
```diff
@@ -184,59 +192,90 @@ export const ApiIcon: React.FC<ApiIconProps> = (props) => {
...
+      } else if (spec.bool) {
+        status.bool = { value: value === true }
+      }
+      toUpdate.status!.inputs!.push(status)
+    }
+  })
+
+  const url = `/proxy/apis/tilt.dev/v1alpha1/uibuttons/${
+    toUpdate.metadata!.name
+  }/status`
```
[the remark this answers is not in the document]
Comment (landism) [reply web/src/ApiButton.tsx:226]: Nope! And I don't want to worry about that in this PR!
1. It's a wider issue than just this url
2. This PR is just moving this code, not changing it.
3. I strongly suspect we'll soon be writing some functions for calling /proxy/apis/tilt.dev
4. I'm not entirely convinced of the benefits of moving one-use urls to consts, but I'm happy to discuss

File: web/src/ApiButton.tsx [hunk 2]
```diff
@@ -184,59 +192,90 @@ export const ApiIcon: React.FC<ApiIconProps> = (props) => {
...
     try {
-      await fetch(url, {
-        method: "PUT",
-        headers: {
-          Accept: "application/json",
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify(toUpdate),
-      })
+      await updateButtonStatus(props.button, inputValues)
```
[the remark this answers is not in the document]
Comment (landism) [reply web/src/ApiButton.tsx:278]: Ideally no but there's always a chance of bugs or weird network stuff.

We historically haven't had a great way to report these errors to the user and some of our other api calls just log errors to the console, but it turns out it's pretty cheap to get it into an error modal. Thanks for pushing on this!

[the remark this answers is not in the document]
Comment (landism) [reply web/src/ApiButton.tsx]: It will now pop up a hud error modal, which we can also use for other errors on failed http requests!

**Comment by landism** [comment]
I've updated the styling to match tooltips and updated the screenshot in this PR's OP to match

**Review by lizzthabet — APPROVED** [review]
i think the styling adjustments make a huge improvement! i tested locally with and without (intentional) errors. there are probably adjustments we could make to that error modal (and probably get analytics on how often it appears?), but that's out of scope here and not a high priority. :)

**Pull request merged**